api/src/controllers/publications.js

```diff
@@ -158,6 +158,8 @@ async function getGoogleScholarPublications(req, res) {
     playwrightConfig.sortBySuffix;
   logger.info(`GScholar profile for user id ${author}: ${url}`);
   let publications = [];
+  let endOfProfile = false;
+  let response;
```

Comment:
Consider initializing the  `response` here, with default values:
```
   response = {
      retrieved: publications.length,
      newPublications: [],
      reachedEnd: endOfProfile,
    };
    ```
    
to avoid some duplication below